Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationFailureRepository.cs

```diff
@@ -103,6 +111,14 @@ public BocListValidationFailureRepository ()
     {
     }
 
+    public int ListFailureCount => _listValidationFailures.Count;
+
+    public int UnhandledListFailureCount => _listValidationFailures.Count(e => !e.Handled);
+
+    public int RowAndCellFailureCount => _rowValidationFailures.Sum(e => e.Value.RowAndCellFailuresCount);
+
+    public int UnhandledRowAndCellFailureCount => _rowValidationFailures.Sum(e => e.Value.UnhandledRowAndCellFailuresCount);
```

Comment:
I have converted the properties into method but have not implemented the caching yet. Wise people once said: There are only two things that are difficult in programming: naming things, cache invalidation, and off-by-one errors.

In this case, cache invalidation gets a bit complex since child container need to report to parent containers to invalidate caches but no such structure exists yet. There is also the fact that unhandled counts are going to be recalculated often anyways since we are handling failures as we go. Currently, we only access each property once and thats it. So any optimization would be for future code.

If you want we can add the complexity and I add caching otherwise, I would leave it as is. Now that the properties are methods nobody can complain that execution takes a while anyways :p

Reply:
Ah, that's true. no caching it is.